src/csv_handler.rs

```diff
@@ -0,0 +1,189 @@
+    let logged_imu = get_data::<T>(component_name);
+
+    if let Some(data) = logged_imu {
+        for entry in data {
+            writter.serialize(&entry.data)?;
+        }
+        writter.flush()?;
+    } else {
+        eprintln!("Logged {component_name} data not found")
+    }
+
+    Ok(())
+}
+
+fn save_log_handle<T: serde::Serialize + Send + Clone + Debug + Sync + 'static>(
+    path: &str,
+    component_name: &str,
+) {
+    match save_log_to_file::<T>(path, component_name) {
+        Ok(_) => eprintln!("{component_name} data saved to {path}"),
+        Err(e) => eprintln!("Error {e}"),
+    }
+}
+
+fn concat_path(component_name: &str) -> String {
+    let file_name = component_name.to_lowercase();
+    let mut path = PathBuf::from(OUTPUT_PATH);
+    path.push(format!("{}.{}", file_name, CSV_EXTENSION));
+    path.to_string_lossy().into_owned()
+}
+
+fn save_imu_log_to_file() {
+    save_log_handle::<Data>(concat_path(IMU_LOG).as_str(), IMU_LOG);
+}
+
+fn save_gps_log_to_file() {
+    save_log_handle::<Data>(concat_path(GPS_LOG).as_str(), GPS_LOG);
+}
+
+fn save_inertial_nav_to_file() {
+    save_log_handle::<Telemetry>(
+        concat_path(INTERTIAL_NAVIGATOR_LOG).as_str(),
+        INTERTIAL_NAVIGATOR_LOG,
+    );
+}
+
+fn save_kalman_log_to_file() {
+    save_log_handle::<Telemetry>(concat_path(KALMAN_LOG).as_str(), KALMAN_LOG);
+}
+
+fn save_general_log_to_file() {
+    save_log_handle::<String>(concat_path(GENERAL_LOG).as_str(), GENERAL_LOG);
+}
+
+fn save_groundtruth_log_to_file() {
+    save_log_handle::<Data>(concat_path(GROUNDTRUTH_LOG).as_str(), GROUNDTRUTH_LOG);
+}
+
+fn save_moving_average_log_to_file() {
+    save_log_handle::<Data>(concat_path(MOVING_AVERAGE_LOG).as_str(), MOVING_AVERAGE_LOG);
+}
+
+pub fn save_logs_to_file() {
+    save_gps_log_to_file();
+    save_imu_log_to_file();
+    save_inertial_nav_to_file();
+    save_kalman_log_to_file();
+    save_general_log_to_file();
+    save_groundtruth_log_to_file();
+    save_moving_average_log_to_file();
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::logger::log;
+    use std::fs;
+    use std::io::Read;
+    use std::time::SystemTime;
+
+    const TEST_PATH: &str = "test_output/test_log.csv";
+    const TEST_NO_DATA_PATH: &str = "test_output_no_data/test_log.csv";
+    const TEST_COMPONENT: &str = "TEST_COMPONENT";
+    const EMPTY_TEST_COMPONENT: &str = "EMPTY_TEST_COMPONENT";
+
+    #[cfg(target_os = "windows")]
+    const PATH_SEPARATOR: char = '\\';
+    
+    #[cfg(not(target_os = "windows"))]
+    const PATH_SEPARATOR: char = '/';
+
+    fn create_test_data() -> Data {
+        Data {
+            x: 1.0,
+            y: 2.0,
+            z: 3.0,
+            timestamp: SystemTime::now(),
+        }
+    }
+
+    fn cleanup_test_directory() {
+        let _ = fs::remove_dir_all("test_output");
+    }
+
+    fn read_file_content(path: &str) -> String {
+        let mut file = fs::File::open(path).expect("Failed to open file");
```

Comment:
Unwrap is sufficient here. Expect should be used where the unwrap shouldn't fail and in expect you provide explanation why. Backtrace from `unwrap()` will give you enough information, so you don't need to print that opening a file failed.
```suggestion
        let mut file = fs::File::open(path).unwrap();
```